include/libtorrent/span.hpp

```diff
@@ -90,6 +90,14 @@ namespace aux {
 		span(Cont& c) // NOLINT
 			: m_ptr(c.data()), m_len(c.size()) {}
 
+		// allow construction from const containers if T is const
+		template <typename Cont
+			, typename U = typename std::remove_reference<decltype(*std::declval<Cont>().data())>::type
+			, typename = typename std::enable_if<aux::compatible_type<U, T>::value
+				&& std::is_const<T>::value>::type>
```

Comment:
Isn't this exactly the same condition as the overload above, except with the ``&& std::is_const<>`` added?
I don't really understand which cases require this overload but doesn't match the existing one. The intention with ``aux::compatible_type`` was to capture all the const conversions correctly.

Reply:
There's a ``test_span.cpp`` which exercises the cases I had thought of. Could you add the missing one(s) there too?